packages/sheet/src/index.js

```diff
@@ -108,27 +108,31 @@ export class StyleSheet {
     }
     const tag = this.tags[this.tags.length - 1]
 
+    if (process.env.NODE_ENV !== 'production') {
+      const isImportRule =
+        rule.charCodeAt(0) === 64 && rule.charCodeAt(1) === 105
+
+      if (isImportRule && (this: any)._alreadyInsertedOrderInsensitiveRule) {
+        // this would only cause problem in speedy mode
+        // but we don't want enabling speedy to affect the observable behavior
+        // so we report this error at all times
+        console.error(
+          `You have attempted inserting such rule:\n` +
+            rule +
+            '\n\n`@import` rules must precede all other types of rules in a stylesheet. Please ensure that you insert them first.'
+        )
+      }
+
+      ;(this: any)._alreadyInsertedOrderInsensitiveRule =
```

Comment:
Not really a blocker but why not make this an optional property on the class?

Reply:
Just a minor difference - an optional property would be visible as part of the public type and we don't need it to be there.